### PR TITLE
Block if grub not installed or config file missing

### PR DESCRIPTION
**Pull request by timmullin**
Case RE-83:

Changelog: Block if the grub2-pc package is not installed
  or if the grub.cfg file is missing.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: lib/Elevate/Blockers/Grub2.pm [hunk 1]
```diff
@@ -99,6 +103,28 @@ sub _blocker_blscfg ($self) {
     return 0;
 }
 
+sub _blocker_grub_not_installed ($self) {
+
+    return 0 if Cpanel::Pkgr::is_installed('grub2-pc');
+
+    return $self->has_blocker( <<~EOS );
+    The grub2-pc package is not installed. The GRUB2 boot loader is
+    required to upgrade via leapp.
```
Comment (cPholloway) [inline lib/Elevate/Blockers/Grub2.pm:112]: Let's advise them to consider reaching out to support if they hit this blocker.  There is already some wording to do this earlier in this file via `_blocker_grub2_workaround()`

File: lib/Elevate/Blockers/Grub2.pm [hunk 2]
```diff
@@ -99,6 +103,28 @@ sub _blocker_blscfg ($self) {
...
+    EOS
+}
+
+sub _blocker_grub_config_missing ($self) {
+
+    if ( !-e '/boot/grub2/grub.cfg' && !-e '/boot/grub/grub.cfg' ) {
+
+        return $self->has_blocker( <<~EOS );
+        The GRUB2 config file is missing.
```
Comment (cPholloway) [inline lib/Elevate/Blockers/Grub2.pm:121]: Let's advise them to consider reaching out to support if they hit this blocker.  There is already some wording to do this earlier in this file via `_blocker_grub2_workaround()`

File: lib/Elevate/Blockers/Grub2.pm [hunk 3]
```diff
@@ -99,6 +103,28 @@ sub _blocker_blscfg ($self) {
...
+
+    return $self->has_blocker( <<~EOS );
+    The grub2-pc package is not installed. The GRUB2 boot loader is
+    required to upgrade via leapp.
+    EOS
+}
+
+sub _blocker_grub_config_missing ($self) {
+
+    if ( !-e '/boot/grub2/grub.cfg' && !-e '/boot/grub/grub.cfg' ) {
```
Comment (cPholloway) [inline lib/Elevate/Blockers/Grub2.pm:118]: We should use `-f` instead of `-e` here.  Possibly `-s` too to make sure it is not a touch file.

**Review by cPholloway — CHANGES_REQUESTED** [review]

**Review by cPholloway — APPROVED** [review]

**Review by toddr — APPROVED** [review]

**Review by annes449 — COMMENTED** [review]
QA looks good.

**Comment by rch77** [comment]
Now it blocks if there is /boot/efi/EFI/centos/grub.cfg only. I think UEFI systems use /boot/efi/EFI/centos/grub.cfg.
I have added symlink from /boot/grub2/ to ../efi/EFI/centos/grub.cfg but I'm not sure if it'll break something else during the process.
OS: Centos 7.9

**Pull request merged**